test/clojure_tic_tac_toe/tic_tac_toe_rules_test.clj

```diff
@@ -0,0 +1,34 @@
+(ns clojure-tic-tac-toe.tic_tac_toe_rules_test
```

Comment:
I like how clean you made these tests.  It's really easy to read.  However, I'm not sure they provide much value.  `get-winning-sets` is just a static data structure.  There is no behavior to test.  Therefore, your tests just end up duplicating the your code.  It would be more valuable to test this through the `has-player-won?` tests.

Reply:
I decided not to test that `has-player-won?` works for each row, column, and diagonal (I don't think it's necessary).

Reply:
You should probably keep those tests as tests for has-player-won?. Those rules are a pretty important part of your program and it would be good to know if they break, right?
If you deleted some of those winning sets, you won't get any test failures. You definitely would know not to delete some of those winning sets but if this project was touched by more people, they might not know about it and modify those winning sets, which could introduce a bug, provide a bad user experience, and make it a lot harder to find that bug

Reply:
Ok, I've tested all the winning sets in `has-player-won?` now.